src/cmap/auth/gssapi.ts

```diff
@@ -174,7 +174,7 @@ function performGssapiCanonicalizeHostName(
   mechanismProperties: MechanismProperties,
   callback: Callback<string>
 ): void {
-  if (!mechanismProperties.gssapiCanonicalizeHostName) return callback(undefined, host);
+  if (!mechanismProperties.CANONICALIZE_HOST_NAME) return callback(undefined, host);
```

Comment:
I did a little test script (below) and this line is reachable with the `gssapiCanonicalizeHostName` option specified. So this isn't dead code, I think we need to continue supporting the old option but we could assign it to the new one inside the credentials class or something? Even though it wasn't in our types it did work in 3.x and still works in 4.x.

```ts
import { MongoClient } from '.';

const client = new MongoClient(
  'mongodb://u@localhost?authMechanism=GSSAPI&authMechanismProperties=SERVICE_NAME:mongodb,gssapiCanonicalizeHostName:true&maxPoolSize=1'
);

async function main(args) {
  await client.connect();
}

main(process.argv)
  .then(console.log)
  .catch(console.error)
  .finally(() => client.close());
```

Reply:
Maybe we should at least mark it deprecated then? We're gearing up for a minor release anyway, shouldn't be a big deal

Reply:
I've put it back and marked as deprecated and added a test to verify it still works when set.